metaflow/plugins/events_decorator.py

```diff
@@ -188,6 +241,63 @@ def flow_init(
 
         # TODO: Handle scenario for local testing using --trigger.
 
+    def format_deploytime_value(self):
+        for trigger in self.triggers:
+            # Case were trigger is a function that returns a list of events
+            # Need to do this bc we need to iterate over list later
+            if isinstance(trigger, DeployTimeField):
+                if isinstance(deploy_time_eval(trigger), list):
```

Comment:
Calling `deploy_time_eval` twice. Also, at the top level, something that is a deploy-time field will be:
  - a string (so needs to be handled like line 96)
  - a dictionary (single event in dict form)
  - a list of the two things above (when it is pushed as events)

So for string, that is done (we create the proper entry and done). For dictionary, we may need further processing of the elements (so continue). For list, add each element as above.